### PR TITLE
7 BOT Transformer cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-game/src/main/java/forge/game/card/CounterEnumType.java [hunk 1]
```diff
@@ -184,6 +184,8 @@ public enum CounterEnumType {
     INFECTION("INFCT", 0, 230, 66),
 
     INGENUITY("INGTY", 67, 186, 205),
+    
+    INTELL("INTEL", 80, 250, 180),
```
Comment (Northmoc) [inline forge-game/src/main/java/forge/game/card/CounterEnumType.java:188]: why `LL` ?

File: forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+Name:Blaster, Combat DJ
+ManaCost:3 R G
+Types:Legendary Artifact Creature Robot
+PT:3/3
+K:More Than Meets the Eye:1 R G
+S:Mode$ Continuous | Affected$ Artifact.Creature+Other+YouCtrl,Vehicles+Other+YouCtrl | AddKeyword$ Modular:1 | Description$ Other nontoken artifact creatures and Vehicles you control have modular 1. (They enter the battlefield with an additional +1/+1 counter on them. When they die, you may put their +1/+1 counters on target artifact creature.)
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt:6]: nontoken

File: forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Oracle:More Than Meets the Eye {1}{R}{G} (You may cast this card converted for {1}{R}{G}.)\nOther nontoken artifact creatures and Vehicles you control have modular 1. (They enter the battlefield with an additional +1/+1 counter on them. When they die, you may put their +1/+1 counters on target artifact creature.)\nWhenever you put one or more +1/+1 counters on Blaster, convert it.
+
+ALTERNATE
+
+Name:Blaster, Morale Booster
+ManaCost:no cost
+Colors:red,green
+Types:Legendary Artifact
+K:Modular:3
+A:AB$ MoveCounter | Cost$ X T | Source$ Self | ValidTgts$ Artifact.Other| TgtPrompt$ Select target artifact to get counters | CounterType$ P1P1 | CounterNum$ X | SorcerySpeed$ True | SubAbility$ DBAnimate | SpellDescription$ Move X +1/+1 counters from Blaster onto another target artifact. That artifact gains haste until end of turn. If Blaster has no +1/+1 counters on it, convert it. Activate only as a sorcery.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt:21]: NICKNAME

File: forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+SVar:DBAnimate:DB$ Animate | Defined$ Targeted | Types$ Artifact | Keywords$ Haste | SubAbility$ DBConvert
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt:22]: probably just use Pump - don't need to make it an artifact

File: forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ALTERNATE
+
+Name:Blaster, Morale Booster
+ManaCost:no cost
+Colors:red,green
+Types:Legendary Artifact
+K:Modular:3
+A:AB$ MoveCounter | Cost$ X T | Source$ Self | ValidTgts$ Artifact.Other| TgtPrompt$ Select target artifact to get counters | CounterType$ P1P1 | CounterNum$ X | SorcerySpeed$ True | SubAbility$ DBAnimate | SpellDescription$ Move X +1/+1 counters from Blaster onto another target artifact. That artifact gains haste until end of turn. If Blaster has no +1/+1 counters on it, convert it. Activate only as a sorcery.
+SVar:DBAnimate:DB$ Animate | Defined$ Targeted | Types$ Artifact | Keywords$ Haste | SubAbility$ DBConvert 
+SVar:DBConvert:DB$ SetState | Mode$ Convert | ConditionPresent$ Card.Self+HasCounters | ConditionCompare$ EQ0
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt:23]: this line will probably need StackDescription – see Arcee

File: forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+DeckHints:Type$Vehicle|Artifact
+Oracle:More Than Meets the Eye {1}{R}{G} (You may cast this card converted for {1}{R}{G}.)\nOther nontoken artifact creatures and Vehicles you control have modular 1. (They enter the battlefield with an additional +1/+1 counter on them. When they die, you may put their +1/+1 counters on target artifact creature.)\nWhenever you put one or more +1/+1 counters on Blaster, convert it.
+
+ALTERNATE
+
+Name:Blaster, Morale Booster
+ManaCost:no cost
+Colors:red,green
+Types:Legendary Artifact
+K:Modular:3
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt:20]: you missed a NICKNAME

File: forge-gui/res/cardsfolder/upcoming/blitzwing_cruel_tormentor_blitzwing_adaptive_assailant.txt [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+ALTERNATE
+
+Name:Blitzwing, Adaptive Assailant
+ManaCost:no cost
+Colors:black
+Types:Legendary Artifact Vehicle
+PT:3/5
+K:Living metal
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigCharm | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of combat on your turn, choose flying or indestructible at random. NICKNAME gains that ability until end of turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/blitzwing_cruel_tormentor_blitzwing_adaptive_assailant.txt:21]: use Pump with "KWChoice" for this

File: forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+A:AB$ MoveCounter | Cost$ X T | Source$ Self | ValidTgts$ Artifact.Other| TgtPrompt$ Select target artifact to get counters | CounterType$ P1P1 | CounterNum$ X | SorcerySpeed$ True | SubAbility$ DBAnimate | SpellDescription$ Move X +1/+1 counters from NICKNAME onto another target artifact. That artifact gains haste until end of turn. If Blaster has no +1/+1 counters on it, convert it. Activate only as a sorcery.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/blaster_combat_dj_blaster_morale_booster.txt:21]: - [ ] spacing
- [ ] TgtPrompt$ should include "other"
- [ ] split this SpellDescription as appropriate to lines 22 and 23 – some of those lines may need a StackDescription too (probably line 23)

**Comment by Northmoc** [comment]
Whoever merges this PR should make sure to Squash merge it

**Comment by Northmoc** [comment]
> Whoever merges this PR should make sure to Squash merge it

Or @Simisays rebase down to a reasonable number of commits

File: forge-gui/res/tokenscripts/ravage.txt [hunk 9]
```diff
@@ -0,0 +1,8 @@
+Name:Ravage
+ManaCost:no cost
+Colors:blue
+Types:Legendary Artifact Creature Robot
+PT:3/3
+K:Menace
+K:Deathtouch
+Oracle:menace, deathtouch
```
Comment (Northmoc) [inline forge-gui/res/tokenscripts/ravage.txt:8]: cap menace

File: forge-gui/res/cardsfolder/upcoming/starscream_power_hungry_starscream_seeker_leader.txt [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+Name:Starscream, Seeker Leader
+ManaCost:no cost
+Colors:black
+Types:Legendary Artifact Vehicle
+K:Living metal
+K:Flying
+K:Menace
+K:Haste
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigMonarch | TriggerZones$ Battlefield | TriggerDescription$ Whenever NICKNAME deals combat damage to a player, if there is no monarch, that player becomes the monarch.
+SVar:TrigMonarch:DB$ BecomeMonarch | Defined$ TriggeredTarget | CheckSVar Monarch | SVarCompare$ EQ0
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/starscream_power_hungry_starscream_seeker_leader.txt:25]: This check shouldn't be working - missing `$`

File: forge-gui/res/cardsfolder/upcoming/starscream_power_hungry_starscream_seeker_leader.txt [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+T:Mode$ BecomeMonarch | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigConvert | TriggerDescription$ Whenever you become the monarch, convert NICKNAME.
+SVar:TrigConvert:DB$ SetState | Mode$ Convert
+SVar:Monarch:PlayerCount.IsMonarch$Amount
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/starscream_power_hungry_starscream_seeker_leader.txt:28]: This seems wrong – did you test this?

File: forge-gui/res/cardsfolder/upcoming/starscream_power_hungry_starscream_seeker_leader.txt [hunk 12]
```diff
@@ -0,0 +1,29 @@
+Name:Starscream, Power Hungry
+ManaCost:3 B
+Types:Legendary Artifact Creature Robot
+PT:2/3
+K:More Than Meets the Eye:2 B
+K:Flying
+T:Mode$ Drawn | ValidCard$ Card.YouCtrl | Condition$ Monarch | TriggerZones$ Battlefield | Execute$ TrigDrain | TriggerDescription$ Whenever you draw a card, if you’re the monarch, target opponent loses 2 life.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/starscream_power_hungry_starscream_seeker_leader.txt:7]: `’` also in Oracle

File: forge-gui/res/cardsfolder/upcoming/flamewar_brash_veteran_flamewar_streetwise_operative.txt [hunk 13]
```diff
@@ -0,0 +1,27 @@
+Name:Flamewar, Brash Veteran
+ManaCost:1 B R
+Types:Legendary Artifact Creature Robot
+PT:3/2
+K:More Than Meets the Eye:B R
+A:AB$ PutCounter | Cost$ Sac<1/Artifact.Other/another artifact> | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SorcerySpeed$ True | SubAbility$ DBConvert | AILogic$ AristocratCounters | SpellDescription$ Put a +1/+1 counter on CARDNAME and convert it.
+SVar:DBConvert:DB$ SetState | Mode$ Convert
+A:AB$ ChangeZone | Cost$ 1 Discard<1/Hand> | ValidCard$ Card.IsRemembered+ExiledWithSource | Origin$ Exile | Destination$ Hand | SpellDescription$ Put all exiled cards you own with intel counters on them into your hand.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/flamewar_brash_veteran_flamewar_streetwise_operative.txt:8]: This ValidCard seems wrong – this should be **any** exiled card you own with intel counters on it... so if somehow some other card managed to do that, this should get those, too

File: forge-gui/res/cardsfolder/upcoming/flamewar_brash_veteran_flamewar_streetwise_operative.txt [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+DeckHints:Ability$Counters
+DeckHas:Ability$Sacrifice|Discard|Counters
+Oracle:More Than Meets the Eye {B}{R} (You may cast this card converted for {B}{R}.)\nSacrifice another artifact: Put a +1/+1 counter on Flamewar and convert it. Activate only as a sorcery.\n{1}, Discard your hand: Put all exiled cards you own with intel counters on them into your hand.
+ALTERNATE
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/flamewar_brash_veteran_flamewar_streetwise_operative.txt:12]: blank line before ALTERNATE pls

File: forge-gui/res/tokenscripts/ravage.txt [hunk 15]
```diff
@@ -0,0 +1,8 @@
+Name:Ravage
+ManaCost:no cost
+Colors:blue
```
Comment (Northmoc) [inline forge-gui/res/tokenscripts/ravage.txt:3]: should be black?

File: forge-gui/res/cardsfolder/upcoming/soundwave_sonic_spy_soundwave_superior_captain.txt [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+Oracle:More Than Meets the Eye {2}{W}{U}{B} (You may cast this card converted for {2}{W}{U}{B}.)\nWhenever one or more creature tokens you control deal combat damage to a player, exile target instant or sorcery card with mana value equal to the damage dealt from their graveyard. Copy it. You may cast the copy without paying its mana cost. If you do, convert Soundwave.
+
+ALTERNATE
+
+Name:Soundwave, Superior Captain
+ManaCost:no cost
+Colors:white,blue,black
+Types:Legendary Artifact
+T:Mode$ SpellCast | ValidCard$ Spell.cmcOdd | ValidActivatingPlayer$ You | Execute$ TrigConvertRavage | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast a spell with an odd mana value, convert NICKNAME. If you do, create Ravage, a legendary 3/3 black Robot artifact creature token with menace and deathtouch
+SVar:TrigConvertRavage:DB$ SetState | Mode$ Convert | SubAbility$ DBTokenRavage
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/soundwave_sonic_spy_soundwave_superior_captain.txt:24]: needs a "RememberChanged" and then a check in line 25 to satisfy "If you do"

File: forge-gui/res/cardsfolder/upcoming/soundwave_sonic_spy_soundwave_superior_captain.txt [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+SVar:DBTokenRavage:DB$ Token | TokenScript$ ravage
+T:Mode$ SpellCast | ValidCard$ Spell.cmcEven | ValidActivatingPlayer$ You | Execute$ TrigConvertLaserbeak | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast a spell with an even mana value, convert NICKNAME.  If you do, create Laserbeak, a legendary 2/2 blue Robot artifact creature token with flying and hexproof.
+SVar:TrigConvertLaserbeak:DB$ SetState | Mode$ Convert | SubAbility$ DBTokenLaserbeak
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/soundwave_sonic_spy_soundwave_superior_captain.txt:27]: needs a "RememberChanged" and then a check in line 28 to satisfy "If you do"

File: forge-gui/res/cardsfolder/upcoming/soundwave_sonic_spy_soundwave_superior_captain.txt [hunk 18]
```diff
@@ -0,0 +1,30 @@
+Name:Soundwave, Sonic Spy
+ManaCost:1 W U B
+Types:Legendary Artifact Creature Robot
+PT:5/4
+K:More Than Meets the Eye:1 W U B
+T:Mode$ DamageDoneOnce | CombatDamage$ True | ValidSource$ Creature.Token+YouCtrl | TriggerZones$ Battlefield | ValidTarget$ Player | Execute$ TrigCast | TriggerDescription$ nWhenever one or more creature tokens you control deal combat damage to a player, exile target instant or sorcery card with mana value equal to the damage dealt from their graveyard. Copy it. You may cast the copy without paying its mana cost. If you do, convert NICKNAME.
+SVar:TrigCast:DB$ ChangeZone | Origin$ Graveyard | Destination$ Exile | TgtPrompt$ Choose target instant or sorcery card in that player' graveyard | ValidTgts$ Instant.cmcLEX+OwnedBy TriggeredTarget,Sorcery.cmcLEX+OwnedBy TriggeredTarget | RememberChanged$ True | SubAbility$ DBPlay
+SVar:DBPlay:DB$ Play | Valid$ Card.IsRemembered | ValidZone$ Exile | Controller$ You | CopyCard$ True | WithoutManaCost$ True | ValidSA$ Spell | Optional$ True | SubAbility$ DBConvert
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/soundwave_sonic_spy_soundwave_superior_captain.txt:8]: right now you are converting whether or not you play the copy

File: forge-gui/res/cardsfolder/upcoming/ratchet_field_medic_ratchet_rescue_racer.txt [hunk 19]
```diff
@@ -0,0 +1,27 @@
+Name:Ratchet, Field Medic
+ManaCost:2 W
+Types:Legendary Artifact Creature Robot
+PT:2/4
+K:More Than Meets the Eye:1 W
+T:Mode$ LifeGained | TriggerZones$ Battlefield | ValidPlayer$ You | OptionalDecider$ True | NoResolvingCheck$ True | Execute$ TrigConvert | TriggerDescription$ Whenever you gain life for the first time each turn, put a +1/+1 counter on NICKNAME.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/ratchet_field_medic_ratchet_rescue_racer.txt:6]: bad Desc

Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/ratchet_field_medic_ratchet_rescue_racer.txt:6]: OptionalDecider$ You

**Comment by Simisays** [comment]
@Northmoc Thanks for all your comments. Will try to make them all work when I'm back from home. I wasn't able to test the monarch condition on the last card yet, though if anyone has any idea how that should be worded feel free to tell me. 

Will also squash this branch.

**Comment by Northmoc** [comment]
> @Northmoc Thanks for all your comments. Will try to make them all work when I'm back from home. I wasn't able to test the monarch condition on the last card yet, though if anyone has any idea how that should be worded feel free to tell me.
> 
> Will also squash this branch.

No problem! In future, I'd appreciate it if you tested before requesting a review – that keeps more of the onus on you as the PR author. Another option is to comment on or mention a script or line that you haven't had the chance to test and aren't sure on.

**Comment by Simisays** [comment]
Something went wrong with the rebase. Will just make a new PR

**Pull request closed without merging**